### ISSUE TITLE
Adding Material2

### ISSUE THREAD

**Issue by andreiolaru** [angular/universal-starter#270]
Hello,

It seems that i am trying to add the material 2 without success. My steps were:

1. Add material reference to package.json, version 2.0.0-alpha.10;, 
`rimraf node_modules`
`npm install`

2. Into app.module.ts:
a. import the MaterialModule (`import { MaterialModule } from '@angular/material';`);
b. add to Imports: `MaterialModule.forRoot()`
3. `run npm start`

Result is the follosing:

`angular-poc\dist\server\index.js:42
/******/                Object.defineProperty(exports, name, {
                               ^

TypeError: Cannot redefine property: b
    at Function.defineProperty (<anonymous>)
    at Function.module.exports.__webpack_require__.d (C:\Projects\angular-poc\dist\server\index.js:42:19)
    at Object.module.exports.exports.b (C:\Projects\angular-poc\dist\server\index.js:45191:64)
    at __webpack_require__ (C:\Projects\angular-poc\dist\server\index.js:21:30)
    at Object.<anonymous> (C:\Projects\angular-poc\dist\server\index.js:66838:73)
    at __webpack_require__ (C:\Projects\angular-poc\dist\server\index.js:21:30)
    at Object.<anonymous> (C:\Projects\angular-poc\dist\server\index.js:65801:66)
    at __webpack_require__ (C:\Projects\angular-poc\dist\server\index.js:21:30)
    at Object.<anonymous> (C:\Projects\angular-poc\dist\server\index.js:61939:18)
    at __webpack_require__ (C:\Projects\angular-poc\dist\server\index.js:21:30)
[nodemon] app crashed - waiting for file changes before starting...`

**Comment by antonyRoberts** [comment]
@andreiolaru This is resolved in the another issue [Cannot redefine property: b](https://github.com/angular/universal-starter/issues/266)

**Comment by MarkPieszak** [comment]
So at the moment Material2 doesn't support Universal yet, but you can track this issue to see when we'll get some traction there. https://github.com/angular/material2/issues/308

**Comment by gdi2290** [comment]
you can manually add support to Material by rewriting some of their services. If you do then you have to include MaterialModule.forRoot() before UniversalModule since they require HttpModule and we have to replace parts of it

**Comment by gdi2290** [comment]
Material still doesn't work. You have to create Node versions of some of their services otherwise you will run into problems with some of the globals they're using. There is also problems with shared state with one of their registries service that needs to be recreated. If someone wants Material support they have to do this for the components they use

**Comment by m98** [comment]
@gdi2290 You [commented](https://github.com/angular/material2/issues/308#issuecomment-247738536) on the Material 2 and said:

> if you're using webpack you can use a string-replace-loader to fix the material2 package by replacing the string value of each file. I'll update the universal-starter to include it as an example

But I did not see those changes in the universal-starter, can you please provide more information or let me know if you changed the universal?

Thanks

**Comment by MarkPieszak** [comment]
It should be able to work with Material now without too many problems @m98 
We can open an issue in the Material repo if there's still something not working.

**Comment by m98** [comment]
@MarkPieszak thanks

I am getting warnings with some of material components (progress bar is one of them) 

I guess I need to open an issue on Material repo.

The serer side rendering works, but Node still logs those warnings.

**Comment by MarkPieszak** [comment]
Yeah open it up there if you can, it's most likely just something in the Component that needs updating on the Material-end. Thanks!
